python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+from __future__ import absolute_import, division, print_function
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from collections import namedtuple
+import numpy as np
+
+# LSST imports
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.meas.base as measBase
+import lsst.afw.table as afwTable
+import lsst.afw.detection as afwDet
+import lsst.pex.exceptions as pexExcept
+import lsst.pex.logging as pexLog
+import lsst.pex.config as pexConfig
+
+__all__ = ("DipoleFitPluginConfig", "DipoleFitTask", "DipoleFitPlugin",
+           "DipoleFitAlgorithm")
+
+
+# Create a new measurement task (`DipoleFitTask`) that can handle all other SFM tasks but can
+# pass a separate pos- and neg- exposure/image to the `DipoleFitPlugin`s `run()` method.
+
+
+class DipoleFitPluginConfig(measBase.SingleFramePluginConfig):
+    """
+    Class to initialize and store dipole fitting configuration parameters
+    """
+
+    maxSeparation = pexConfig.Field(
+        dtype=float, default=5.,
+        doc="Assume dipole is not separated by more than maxSeparation * psfSigma")
+
+    relWeight = pexConfig.Field(
+        dtype=float, default=0.5,
+        doc="""Relative weighting of pre-subtraction images (higher -> greater influence of pre-sub.
+        images on fit)""")
+
+    tolerance = pexConfig.Field(
+        dtype=float, default=1e-7,
+        doc="Fit tolerance")
+
+    fitBgGradient = pexConfig.Field(
+        dtype=bool, default=True,
+        doc="Include parameters to fit for linear gradient in pre-sub. images")
+
+    fitSeparateNegParams = pexConfig.Field(
+        dtype=bool, default=False,
+        doc="Include parameters to fit for negative values (flux, gradient) separately from pos.")
+
+    # Config params for classification of detected diaSources as dipole or not
```

Comment:
All documentation for config parameters really, really, really should be in the doc strings themselves. Users should never be expected to read source code to understand config parameters.